core/equations.gms

```diff
@@ -838,16 +832,65 @@ q_limitCO2(ttot+1,regi) $((pm_ttot_val(ttot+1) ge max(cm_startyear,2055)) AND (t
          =l=
          vm_emiTe(ttot,regi,"co2");
 
+***---------------------------------------------------------------------------
+*' Adjustment costs - calculation of the relative change to last time step 
+***---------------------------------------------------------------------------
+
 q_eqadj(regi,ttot,teAdj(te))$(ttot.val ge max(2010, cm_startyear)) ..
-         v_adjFactor(ttot,regi,te)
-         =e=
-         power(
-         (sum(te2rlf(te,rlf),vm_deltaCap(ttot,regi,te,rlf)) - sum(te2rlf(te,rlf),vm_deltaCap(ttot-1,regi,te,rlf)))/(pm_ttot_val(ttot)-pm_ttot_val(ttot-1))
-         ,2)
-                /( sum(te2rlf(te,rlf),vm_deltaCap(ttot-1,regi,te,rlf)) + p_adj_seed_reg(ttot,regi) * p_adj_seed_te(ttot,regi,te)  
-                   + p_adj_deltacapoffset("2010",regi,te)$(ttot.val eq 2010) + p_adj_deltacapoffset("2015",regi,te)$(ttot.val eq 2015)
-                   + p_adj_deltacapoffset("2020",regi,te)$(ttot.val eq 2020) + p_adj_deltacapoffset("2025",regi,te)$(ttot.val eq 2025)
-                  );
+  v_adjFactor(ttot,regi,te)
+  =e=
+  power(
+    ( sum(te2rlf(te,rlf), vm_deltaCap(ttot,regi,te,rlf)) - sum(te2rlf(te,rlf), vm_deltaCap(ttot-1,regi,te,rlf)) )
+    / ( pm_ttot_val(ttot) - pm_ttot_val(ttot-1) )
+  , 2)
+  / ( sum(te2rlf(te,rlf), vm_deltaCap(ttot-1,regi,te,rlf)) + p_adj_seed_reg(ttot,regi) * p_adj_seed_te(ttot,regi,te)  
+      + p_adj_deltacapoffset("2010",regi,te)$(ttot.val eq 2010) + p_adj_deltacapoffset("2015",regi,te)$(ttot.val eq 2015)
+      + p_adj_deltacapoffset("2020",regi,te)$(ttot.val eq 2020) + p_adj_deltacapoffset("2025",regi,te)$(ttot.val eq 2025)
+    )
+;
+
+***---------------------------------------------------------------------------
+*' Calculate changes to REF in cm_startyear - needed to limit them via refunded adj costs 
+***---------------------------------------------------------------------------
+*' calculating the absolute change of output with respect to the value in REF for each te (counting SE, FE, UE and CCS)
+q_changeProdStartyear(t,regi,te)$( (t.val gt 2005) AND (t.val eq cm_startyear ) )..
+  v_changeProdStartyear(t,regi,te)
```

Comment:
this is very minor, but as by default you only want to define v_changeProdStartyear to a single year always, you could remove the time dimension from the variable and equation, and use a sum to control the time dimension in vm_prodSe, vm_prodFE, ...
It wouldn't change in any way the way your formulation works, but it would reduce unnecessary dimensions from memory allocation and gdx files.
This is very minor in this case, but I just wanted to mention to keep in mind for alternative formulations in the future.

Reply:
right. I didn't think about this but just automatically added the time dimension (while trying to make sure that only one entry will ever be filled in a run), but you are right - especially if one loads other gdxes from runs fixed until other times, there might be residual information carried over in the gdx, which would be confusing...

I will check if I can remove the time dimension today